src/Sets/Star.jl

```diff
@@ -166,3 +167,45 @@ Return the predicate of a star.
 A polyhedral set representing the predicate of the star.
 """
 predicate(X::STAR) = set(X)
+
+function _intersection!(c, V, P::Union{HPoly, HPolygon}, H::HalfSpace)
+    a′ = transpose(V) * H.a
+    b′ = H.b - dot(H.a, c)
+    H′ = HalfSpace(a′, b′)
+    return addconstraint!(P, H′)
+end
+
+function intersection!(X::STAR, H::HalfSpace)
+    _intersection!(center(X), basis(X), predicate(X), H)
```

Comment:
Can we rename this method to `_intersection_star!`? There is no dispatch on a `STAR` anymore and it might get misused.

Reply:
sounds better yes